### PR TITLE
feat(Flyout): Add as prop for Flyout title

### PR DESCRIPTION
**Pull request by sooinc**
### Overview

- Adds optional `as` prop for Flyout title heading-level

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: [ABC-123]
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change
- [ ] This PR includes testing instructions tests for the code change
- [ ] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### Testing Instructions

1. See instructions in the Portal PR (link below).

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/mono/pull/7325)  | [Portal Env](https://tayra.codecademy.com/workspaces/new?PR_ENV=le-pr-7325)   |

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@56.1.4-alpha.f14fb6.0<br/>@codecademy/gamut-kit@0.6.435-alpha.f14fb6.0<br/>@codecademy/styleguide@67.1.5-alpha.f14fb6.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://66b6205246bbbc7ef37c7bcd--gamut-preview.netlify.app>https://66b6205246bbbc7ef37c7bcd--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/66b6205246bbbc7ef37c7bcd>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@56.1.4-alpha.371dcc.0<br/>@codecademy/gamut-kit@0.6.435-alpha.371dcc.0<br/>@codecademy/styleguide@67.1.5-alpha.371dcc.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@57.0.1-alpha.fe5ea4.0<br/>@codecademy/gamut-kit@0.6.440-alpha.fe5ea4.0<br/>@codecademy/styleguide@68.0.1-alpha.fe5ea4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://66d1c9c0320e5e94ad0ea3a9--gamut-preview.netlify.app>https://66d1c9c0320e5e94ad0ea3a9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/66d1c9c0320e5e94ad0ea3a9>Deploy Logs</a></p>

File: packages/gamut/src/Flyout/index.tsx [hunk 1]
```diff
@@ -72,7 +72,7 @@ export const Flyout: React.FC<FlyoutProps> = ({
             maxWidth="100%"
             justifyContent="space-between"
           >
-            <Text as="h2" fontSize={22}>
+            <Text as="h1" fontSize={22}>
```
Comment (dreamwasp) [inline packages/gamut/src/Flyout/index.tsx:75]: do you think it makes sense for this to be configurable and default to h1?

Comment (sooinc) [reply packages/gamut/src/Flyout/index.tsx:75]: hmm i had that initially but i couldn't think of a case when `title` wont be the first in the heading level

Comment (dreamwasp) [reply packages/gamut/src/Flyout/index.tsx:75]: yeah that makes sense - we can swap to h1 and just refactor if there are any cases of it needing another title level

**Review by dreamwasp — APPROVED** [review]

**Pull request merged**